modules/core/src/cuda_gpu_mat.cpp

```diff
@@ -407,6 +407,22 @@ void cv::cuda::syncOutput(const GpuMat& dst, OutputArray _dst, Stream& stream)
 #endif
 }
 
+void cv::cuda::detachOutput(GpuMat& src, const GpuMat& dst, Stream& stream)
+{
+#ifndef HAVE_CUDA
+    CV_UNUSED(src);
+    CV_UNUSED(dst);
+    CV_UNUSED(stream);
+    throw_no_cuda();
+#else
+    CV_UNUSED(stream);
+    if (src.refcount == dst.refcount)
+    {
+        src = src.clone();
```

Comment:
IMHO, Performance warning must be emitted. Users can see it and update application code.

You may use this "once" emitter: `CV_LOG_ONCE_WARNING()`
Also caller function name should be used in the message.

Reply:
https://github.com/opencv/opencv/commit/58dcd52e37783bc310e5b607cba08c0cb662da62 tried to resolve this review comment, however: https://github.com/opencv/opencv/pull/17863#issuecomment-661536808